src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -892,6 +892,17 @@ public void handleStorageException(
       SnowflakeFileTransferAgent.throwJCEMissingError(operation, ex);
     }
 
+    // If there is no space left in the download location, java.io.IOException is thrown.
+    // Don't retry.
+    if (SnowflakeUtil.getRootCause(ex).getMessage().equals("No space left on device")) {
```

Comment:
Can you put "No space left on device" as constant string in a common file for AWS, Azure, and GCS? so we can maintain easily if any changes needed in future.